casanovo/casanovo.py

```diff
@@ -371,6 +373,17 @@ def setup_model(
                 "model weights"
             ) from None
 
+    # Download model from URL if model is a valid url
+    is_url = _is_valid_url(model)
+    if (model is not None) and is_url:
```

Comment:
**suggestion:** I'd make the flow of the code more explicit here. If the model is retrieved from the cache in the previous lines of code, it won't be `None` anymore. But this code still won't be applicable., so I suggest refactoring the code a bit to avoid confusion.

Reply:
Sounds good, I streamlined the model resolution logic.